src/Workspaces/CoreTestUtilities/ExportProviderCache.cs

```diff
@@ -19,24 +19,24 @@ public static class ExportProviderCache
 
         // Cache the catalog and export provider factory for MefHostServices.DefaultAssemblies
         private static readonly ComposableCatalog s_defaultHostCatalog =
-            CreateAssemblyCatalog(MefHostServices.DefaultAssemblies);
+            CreateAssemblyCatalogNoCache(MefHostServices.DefaultAssemblies);
 
         private static readonly IExportProviderFactory s_defaultHostExportProviderFactory =
-            CreateExportProviderFactory(s_defaultHostCatalog);
+            CreateExportProviderFactoryNoCache(s_defaultHostCatalog);
 
         // Cache the catalog and export provider factory for DesktopMefHostServices.DefaultAssemblies
         private static readonly ComposableCatalog s_desktopHostCatalog =
-            CreateAssemblyCatalog(DesktopMefHostServices.DefaultAssemblies);
+            CreateAssemblyCatalogNoCache(DesktopMefHostServices.DefaultAssemblies);
 
         private static readonly IExportProviderFactory s_desktopHostExportProviderFactory =
-            CreateExportProviderFactory(s_desktopHostCatalog);
+            CreateExportProviderFactoryNoCache(s_desktopHostCatalog);
 
         // Cache the catalog and export provider factory for RoslynServices.RemoteHostAssemblies
         private static readonly ComposableCatalog s_remoteHostCatalog =
-            CreateAssemblyCatalog(RoslynServices.RemoteHostAssemblies);
+            CreateAssemblyCatalogNoCache(RoslynServices.RemoteHostAssemblies);
 
         private static readonly IExportProviderFactory s_remoteHostExportProviderFactory =
-            CreateExportProviderFactory(s_remoteHostCatalog);
+            CreateExportProviderFactoryNoCache(s_remoteHostCatalog);
 
         private static bool _enabled;
```

Comment:
Can't stick a comment on line 47 so doing it here: can we rename the setting property to something more like "Enabled_OnlyCallFromUseExportProvider"? I read the property as "enabled via the attribute" which implies there are *other* ways you can enable that are also valid. I didn't read it as "only use from there".

Reply:
➡️ Name is improved now